src/compiler/compiler.ts

```diff
@@ -70,14 +73,26 @@ export function compile(spec, theme?) {
 
   // FIXME replace FIT with appropriate mechanism once Vega has it
   const FIT = 1;
+
   // TODO: change type to become VgSpec
-  var output = {
+  var output = extend(
+    spec.name ? {name: spec.name} : {},
+    {
       width: layout.width.field ? FIT : layout.width,
       height: layout.height.field ? FIT : layout.height,
-      padding: 'auto',
+      padding: 'auto'
+    },
+    ['viewport', 'background', 'scene'].reduce(function(topLevelConfig, property) {
```

Comment:
I disagree.  The goal of this chunk of code is to optimize order of the vega output. 
While stylistically a little uglier, putting it like this allows you to determine order of the output in one glance.